File: cf886div4/e.cpp
#include<iostream>
#define int unsigned long long
using namespace std;
const int N = 2e5 + 1;
int s[N];
int n, c;
int sum;
int ssum;

bool check(int x) {
    if (4ull * n * x * x + 4ull * sum * x + ssum >= c) {
        return 1;
    }
    else {
        return 0;
    }
}

int solve(int l, int r) {
    if (l >= r) {
        return l;
    }
    int mid = (l + r) / 2;
    if (check(mid)) {
        return solve(l, mid);
    }
    else {
        return solve(mid + 1ll, r);
    }
}

signed main() {
    int t;
    cin >> t;
    while (t--) {
        sum = 0;
        cin >> n >> c;
        for (int i = 1;i <= n;i++) {
            cin >> s[i];
            sum += s[i];
            ssum += s[i] * s[i];
        }
        cout << solve(1, 2e9) << endl;
    }
    return 0;
}

File: cf886div4/e.py
t = int(input())
for i in range(0, t):
    [n, k] = list(map(int, input().split(" ")))
    a = list(map(int, input().split(" ")))
    sum = 0
    ssum = 0
    for i in a:
        sum += i
        ssum += i*i
    l = 1
    r = 2e9
    r = int(r)
    l = int(l)
    ans = 0
    while 1:
        if (l >= r):
            ans = l
            break
        mid = int((l+r)/2)
        if mid*mid*4*n+4*sum*mid+ssum >= k:
            r = mid
        else:
            l = mid+1
    print(ans)
